composition/composition_handler.go

```diff
@@ -83,31 +81,83 @@ func (agg *CompositionHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 		}
 	}
 
-	status := 200
+	status := agg.extractStatusCode(results, w, r)
```

Comment:
Also this function name is missleading, because the basic work it does is to copy the headers.
Maybe the two things: Extraction of the status code and copy the header should be done in separate steps.